### PR TITLE
Matching specific Mocks over generic Mocks

### PR DESCRIPTION
**Pull request by AvdLee**
File Extensions `Mock`s are ignored if a specific URL `Mock` exists.

### CODE REVIEW

File: MockerTests/MockerTests.swift [hunk 1]
```diff
@@ -63,6 +63,30 @@ final class MockerTests: XCTestCase {
         waitForExpectations(timeout: 10.0, handler: nil)
     }
 
+    /// It should ignore file extension mocks if a specific URL is mocked.
+    func testSpecificURLOverGenericMocks() {
+        let expectation = self.expectation(description: "Data request should succeed")
+        let originalURL = URL(string: "https://www.wetransfer.com/sample-image.png")!
```
Comment (ghost) [inline MockerTests/MockerTests.swift:69]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/WeTransfer/Mocker/blob/e5932510fa10a2ef69100924600a2913f58057ae/MockerTests/MockerTests.swift#L69"></span>Force unwrapping should be avoided.</td>
    </tr>
  </tbody>
</table>

File: MockerTests/MockerTests.swift [hunk 2]
```diff
@@ -63,6 +63,30 @@ final class MockerTests: XCTestCase {
...
+        Mock(fileExtensions: "png", dataType: .imagePNG, statusCode: 400, data: [
+            .get: Data()
+        ]).register()
+        Mock(url: originalURL, ignoreQuery: true, dataType: .imagePNG, statusCode: 200, data: [
+            .get: MockedData.botAvatarImageFileUrl.data
+        ]).register()
+
+        URLSession.shared.dataTask(with: originalURL) { (data, _, error) in
+            XCTAssert(error == nil)
+            let image: UIImage = UIImage(data: data!)!
```
Comment (ghost) [inline MockerTests/MockerTests.swift:80]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/WeTransfer/Mocker/blob/e5932510fa10a2ef69100924600a2913f58057ae/MockerTests/MockerTests.swift#L80"></span>Force unwrapping should be avoided.</td>
    </tr>
  </tbody>
</table>

File: MockerTests/MockerTests.swift [hunk 3]
```diff
@@ -63,6 +63,30 @@ final class MockerTests: XCTestCase {
...
+            let sampleImage: UIImage = UIImage(contentsOfFile: MockedData.botAvatarImageFileUrl.path)!
```
Comment (ghost) [inline MockerTests/MockerTests.swift:81]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/WeTransfer/Mocker/blob/e5932510fa10a2ef69100924600a2913f58057ae/MockerTests/MockerTests.swift#L81"></span>Force unwrapping should be avoided.</td>
    </tr>
  </tbody>
</table>

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 11 tests, with 0 failures (0 unexpected) in 9.234 (9.244) seconds</td>
    </tr>
  </tbody>
</table>

## Current coverage for Mocker is `94.81%`
Files changed | - | - 
--- | --- | ---
Mocker.swift | `100.00%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by Boris-Em — APPROVED** [review]

**Pull request merged**